client/poke.go

```diff
@@ -0,0 +1,114 @@
+// Package httpclient provides a testing enviroment to test poke API server.
+package httpclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff"
+)
+
+// GetPokemonByName mimics a user performing a request to get a certain pokemon from the server
+func (c *Client) GetPokemonByName(ctx context.Context, pokemonName string) (Pokemon, error) {
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.URL+c.endPoint+"/"+pokemonName, nil)
+
+	if err != nil {
+		return Pokemon{}, err
```

Comment:
it is good to add more context to your error, `fmt.Errorf("failed to create request: %w", err)`